### PR TITLE
fix: update node to v20, react to v18, and introduce vite

### PR DESCRIPTION
**Pull request by rmenner**
# Alaska Airlines Pull Request

Temp surge preview link: https://auro-docs-site-major-update.surge.sh

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [x] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Update project to use Node.js 20, React 18, and introduce Vite as the build tool. Refactor codebase to use modern React Router practices and improve markdown rendering. Update build and deployment configurations to align with the new Node.js version.

New Features:
- Introduce Vite as the new build tool for the project.

Enhancements:
- Refactor import paths to use aliases for cleaner and more maintainable code.
- Update React Router to use the new 'element' prop instead of 'component' for route definitions.
- Replace 'marked' with 'react-markdown' and associated plugins for better markdown rendering.

Build:
- Update Node.js version to 20.x in the build configuration.
- Switch build output directory from 'build' to 'dist'.

CI:
- Update CI configuration to use Node.js 20.x.

Deployment:
- Update deployment configuration to use Node.js 20.x.

Documentation:
- Update README to reflect changes in environment variable usage for GitHub API tokens.

### CODE REVIEW

**Comment by rmenner** [comment]
@sourcery-ai review

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ To run the Github data service pages, you will need to add a `/.env` file, at th
 In your local `/.env` file add the following: 
 
 ```
-REACT_APP_GITHUB_API_KEY=[add your PAT]
+VITE_GH_TOKEN=[add your PAT]
```
[the remark this answers is not in the document]
Comment (rmenner) [reply README.md:23]: Moved the description to the instructions above.

**Comment by rmenner** [comment]
@sourcery-ai review

File: src/App.js [hunk 2]
```diff
@@ -243,8 +243,8 @@ import AuroBadgeReleases from './content/dynamic/badge/releases';
 // banner
 import AuroBanner from './content/dynamic/banner/component';
 import AuroBannerApi from './content/dynamic/banner/api';
-import AuroBannerInstall from 'content/dynamic/banner/install';
-import AuroBannerReleases from 'content/dynamic/banner/releases';
+// import AuroBannerInstall from 'content/dynamic/banner/install';
+// import AuroBannerReleases from 'content/dynamic/banner/releases';
```
Comment (jason-capsule42) [inline src/App.js:247]: These need to be re-enabled

File: src/functions/renderComponentPage.js [hunk 3]
```diff
@@ -334,17 +335,19 @@ class AuroComponentContent extends MarkdownPageWrapper {
...
           <Query query={RELEASES}>
-            {({ loading, error, data }) => {
+            {({loading, error, data}) => {
+
               if (loading) return <auro-loader laser onlight></auro-loader>;
               if (error) return <p>We are unable to connect to GitHub at the moment, please try back later.</p>;
-
-              return data.organization.team.repositories.nodes.map(({ name, releases }) => (
+  
+              return data.organization.team.repositories.nodes.map(({ name, releases, index }) => (
```
Comment (jason-capsule42) [inline src/functions/renderComponentPage.js:343]:
```suggestion
              return data.organization.team.repositories.nodes.map(({ name, releases }, index) => (
```

Try this?

File: src/functions/renderComponentPage.js [hunk 4]
```diff
@@ -334,17 +335,19 @@ class AuroComponentContent extends MarkdownPageWrapper {
...
                 releases.nodes.length > 0
-                  ? <Release key={name} name={name} releases={releases.nodes} />
+                  ? <Release key={index} name={name} releases={releases.nodes} />
```
Comment (jason-capsule42) [inline src/functions/renderComponentPage.js:345]: Should we concatenate the index to guarantee it's unique?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/content/dynamic/wcss/wcss.js [hunk 5]
```diff
@@ -24,14 +27,14 @@ class AuroButtonInstall extends MarkdownPageWrapper {
...
-          <ReactMarkdown
-            source={this.state.contentBuild}
-            escapeHtml={false}
-            renderers={{
-              code: CodeBlock,
+          {/* <ReactMarkdown
+            children={this.state.contentBuild}
+            remarkPlugins={[remarkGfm,remarkRehype]}
+            rehypePlugins={[rehypeHighlight,rehypeRaw]}
+            components={{
```
[the remark this answers is not in the document]
Comment (rmenner) [reply src/content/dynamic/wcss/wcss.js:34]: In this case the whole page is not getting rendered. I will remove the file.

File: src/content/dynamic/wcss/wcss.js [hunk 6]
```diff
@@ -1,21 +1,24 @@
 import React from "react";
 import { Nav } from './nav';
-import ReactMarkdown from 'react-markdown';
-import CodeBlock from 'components/CodeBlock';
-import { MarkdownPageWrapper } from 'components/markdownPageWrapper';
+// import ReactMarkdown from 'react-markdown';
```
[the remark this answers is not in the document]
Comment (rmenner) [reply src/content/dynamic/wcss/wcss.js:3]: In this case the whole page is not getting rendered. I will remove the file.

File: src/content/dynamic/wcss/wcss.js [hunk 7]
```diff
@@ -1,21 +1,24 @@
...
+import remarkGfm from 'remark-gfm'
+import remarkRehype from "remark-rehype";
+import rehypeRaw from "rehype-raw";
+// import rehypeHighlight from "rehype-highlight";
```
[the remark this answers is not in the document]
Comment (rmenner) [reply src/content/dynamic/wcss/wcss.js:7]: In this case the whole page is not getting rendered. I will remove the file.

File: src/content/dynamic/wcss/wcss.js [hunk 8]
```diff
@@ -1,21 +1,24 @@
...
+import { MarkdownPageWrapper } from '~/components/markdownPageWrapper';
 
-const markdownContent = 'https://raw.githubusercontent.com/AlaskaAirlines/WebCoreStyleSheets/master/README.md';
+// const markdownContent = 'https://raw.githubusercontent.com/AlaskaAirlines/WebCoreStyleSheets/master/README.md';
```
[the remark this answers is not in the document]
Comment (rmenner) [reply src/content/dynamic/wcss/wcss.js:10]: In this case the whole page is not getting rendered. I will remove the file.

File: src/content/dynamic/wcss/wcss.js [hunk 9]
```diff
@@ -1,21 +1,24 @@
...
-      })
-    })
-  }
+  // UNSAFE_componentWillMount() {
+  //   fetch(markdownContent).then((response) => response.text()).then((text) => {
+  //     this.setState({
+  //       contentBuild: text
+  //     })
+  //   })
+  // }
```
[the remark this answers is not in the document]
Comment (rmenner) [reply src/content/dynamic/wcss/wcss.js:21]: In this case the whole page is not getting rendered. I will remove the file.

File: src/App.js (hunk 2 above)
Comment (rmenner) [reply src/App.js]: Banner pages now work

File: src/functions/renderComponentPage.js (hunk 3 above)
Comment (rmenner) [reply src/functions/renderComponentPage.js]: refactored all the map keys to use index

File: src/functions/renderComponentPage.js (hunk 4 above)
Comment (rmenner) [reply src/functions/renderComponentPage.js]: refactored all the map keys to use index + modifier

**Comment by rmenner** [comment]
@sourcery-ai review

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**